test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -1245,6 +1245,75 @@ static public void testCreateLargeObjectAndMegaRef() throws Throwable {
 		Object megaObjectRef = createAssorted(makeMegaObjectRef, megaFields);
 		checkFieldAccessMHOfAssortedType(megaGetterAndSetter, megaObjectRef, megaFields, false);
 	}
+
+	/*
+	 * Create a value type and read the fields before
+	 * they are set. The test should Verify that the
+	 * flattenable fields are set to the default values.
+	 * NULL should never be observed for Qtypes.
+	 */
+	@Test(priority=4)
+	static public void testDefaultValues() throws Throwable {
+
+		/* Test with Qvalue object */
+		String fields[] = {
+				"val1:QValueObject;:value",
+				"val2:QValueObject;:value",
+				"val3:QValueObject;:value",
+				"val4:QValueObject;:value"};
+
+		Class valueClass = ValueTypeGenerator.generateValueClass("object", fields);
+		MethodHandle makeValue = lookup.findStatic(valueClass, "makeValueGeneric", MethodType.methodType(valueClass, Object.class, Object.class, Object.class, Object.class));
+		Object ObjectValue = createAssorted(makeValue, fields);
+		/*
+		* Getters are created in array getterAndWither[i][0] according to the order of fields i
+		* Withers are created in array getterAndWither[i][1] according to the order of fields i
+		*/
+		MethodHandle[][] getterAndWither = generateGenericGetterAndWither(valueClass, fields);
+		for (int i = 0; i < 4; i++) {
+			assertNotNull(getterAndWither[i][0].invoke(ObjectValue));
+		}
+
+		/* Test with assorted value object with long alignment */
+		String assortedValueWithLongAlignmentFields[] = {
+				"point:QPoint2D;:value",
+				"line:QFlattenedLine2D;:value",
+				"o:QValueObject;:value",
+				"l:QValueLong;:value",
+				"d:QValueDouble;:value",
+				"i:QValueInt;:value",
+				"tri:QTriangle2D;:value"};
+		Class assortedValueWithLongAlignmentClass = ValueTypeGenerator.generateValueClass("newAssortedValueWithLongAlignment", assortedValueWithLongAlignmentFields);
+		MethodHandle makeAssortedValueWithLongAlignment = lookup.findStatic(assortedValueWithLongAlignmentClass,
+				"makeValueGeneric", MethodType.methodType(assortedValueWithLongAlignmentClass, Object.class,
+						Object.class, Object.class, Object.class, Object.class, Object.class, Object.class));
+
+		Object assortedValueWithLongAlignment = createAssorted(makeAssortedValueWithLongAlignment, assortedValueWithLongAlignmentFields);
+		MethodHandle[][] assortedValueGetterAndWither = generateGenericGetterAndWither(assortedValueWithLongAlignmentClass, assortedValueWithLongAlignmentFields);
+		for (int i = 0; i < 7; i++) {
+			assertNotNull(assortedValueGetterAndWither[i][0].invoke(assortedValueWithLongAlignment));
```

Comment:
you need to find out why this isnt working, either there is a mistake with the variation or there is something wrong with this test